x-pack/packages/kbn-elastic-assistant/impl/assistant/api/conversations/conversations.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OpenAiProviderType } from '@kbn/stack-connectors-plugin/common/openai/constants';
+import { HttpSetup, IToasts } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import {
+  ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL,
+  ELASTIC_AI_ASSISTANT_API_CURRENT_VERSION,
+} from '@kbn/elastic-assistant-common';
+import { Conversation, Message } from '../../../assistant_context/types';
+
+export interface GetConversationByIdParams {
+  http: HttpSetup;
+  id: string;
+  toasts?: IToasts;
+  signal?: AbortSignal | undefined;
+}
+
+/**
+ * API call for getting conversation by id.
+ *
+ * @param {Object} options - The options object.
+ * @param {HttpSetup} options.http - HttpSetup
+ * @param {string} options.id - Conversation id.
+ * @param {IToasts} [options.toasts] - IToasts
+ * @param {AbortSignal} [options.signal] - AbortSignal
+ *
+ * @returns {Promise<Conversation | undefined>}
+ */
+export const getConversationById = async ({
+  http,
+  id,
+  signal,
+  toasts,
+}: GetConversationByIdParams): Promise<Conversation | undefined> => {
```

Comment:
Consider re-throwing errors (after showing the toast) instead of returning `undefined` for the error case in this function, and all the other APIs in this file (`createConversation`, `deleteConversation`, etc) that return a `Promise<T | undefined>`

Reply:
fixed